adds velocity drive mode & WPI competition changes

brhecker opened the pull request:
This branch contains the changes needed for the PID velocity drive mode. It replaces driving in brake mode for balancing.
The branch also has some tuning changes made for the WPI competition as we started working out of this branch in order to use velocity drive.

brhecker committed:
adds velocity PID control mode to drive

This commit adds the VelocityDriveSparkMax so we can set a specific rpm for the drive to move at.
It also adds supporting mode switching and commands for the drive subsystem.

brhecker committed:
updates VelocityDrive with bug fixes and improvements

This commit adds default pid tuning for the velocity drive, and sets up constraints for closed loop mode.
It also adds new tank drive methods that allow for bypassing input squaring and setting the drive speed by rpm.

JackTWing committed:
Merge branch 'main' into bhecker-velocity-pid-drive

JackTWing committed:
updates Velocity drive

This commit fixed an issue where drive pid values are overwritten.
It also changes the toggle brake mode button to toggle drive mode

JackTWing committed:
adds Comp changes from WPI

snowpuppy reviewed (changes requested):
Overall, I love it. The one thing that I really want to see changed is the removal of the print statements. The print statements are going to get called a lot, which means the serial port will get spammed.

Question: Does the DriveFixedSpeed ever get used anywhere outside the dashboard? It doesn't look like it.

brhecker commented on src/main/java/frc/robot/commands/Drive/DriveFixedSpeed.java:
```diff
@@ -0,0 +1,51 @@
+import frc.robot.subsystems.Drive;
+import frc.robot.testingdashboard.TestingDashboard;
+
+public class DriveFixedSpeed extends CommandBase {
```
The DriveFixedSpeed command can probably just be removed. It was just a quick thing intended for testing and tuning PID loops. Its not used anywhere

brhecker commented on src/main/java/frc/robot/helpers/VelocityDriveSparkMax.java:
```diff
@@ -0,0 +1,74 @@
+        if(m_driveMode == DriveMode.kPIDVelocity)
+        {
+            double rpm = speed * Constants.DRIVE_MAX_MOTOR_RPM;
+            System.out.println("Driving in Velocity PID mode at " + rpm + "rpm, from speed "+ speed + ", P:"+m_PidController.getP());
```
Ah, yes. This was just for debugging purposes, as we had an issue with PID values not getting set when we expected them to be. Fully agree it should be removed

brhecker commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -245,6 +257,33 @@ public void toggleIdleMode() {
+  public void togglePIDDriveMode() {
+    if(m_driveMode == DriveMode.kPIDVelocity){
+      setDriveMode(DriveMode.kPower);
+    }
```
Okay, I will keep that in mind

brhecker committed:
removes printing and fixes minor issues

This commit removes a debugging print statment that is called very often.
It also changes if else syntax to fit consistent style
